id-repository/credential-request-generator/src/main/java/io/mosip/credential/request/generator/util/RestUtil.java

```diff
@@ -209,12 +210,11 @@ public RestTemplate getRestTemplate() throws KeyManagementException, NoSuchAlgor
             try {
                 HttpEntity<Object> httpEntity = (HttpEntity<Object>) requestType;
                 HttpHeaders httpHeader = httpEntity.getHeaders();
-                Iterator<String> iterator = httpHeader.keySet().iterator();
-                while (iterator.hasNext()) {
-                    String key = iterator.next();
-                    if (!(headers.containsKey("Content-Type") && key == "Content-Type"))
-                        headers.add(key, httpHeader.get(key).get(0));
-                }
+				for (String key : httpHeader.keySet()) {
+					String contentType = "Content-Type";
+					if (!(headers.containsKey(contentType) && key.equals(contentType)))
+						headers.add(key, Objects.requireNonNull(httpHeader.get(key)).get(0));

```

Comment:
Objects.requireNonNull should not be used which will throw NPE. instead, if condition for null check to be used.